### ISSUE TITLE
Issues with version control

### ISSUE THREAD

**Issue by zme1** [obdurodon/dh_course#136]
So today I've both discovered and created problems within our schema. All of them are fixable and I've been updating the document as I need to make changes, but should I be saving multiple versions of my edits instead of overwriting it? I don't think it's an issue that I've over written it, just because GitHub shows the edits I've made, but I want to make sure before I do any more editing.

**Comment by zme1** [comment]
tl;dr: Save a backup under a different filename if you want to be able to compare your current version to an earlier one easily, but you don't have to do it for backup purposes because GitHub backs up everything automatically.

GitHub saves every "commit", and you can always unwind (and then rewind) subsequent edits, so there's no need to make explicit backups. On the other hand, if you want to compare the old and new versions as complete texts, it's easier to do that if you've saved the file under two different names.

GitHub also has a mechanism called "branching" that lets you create parallel development tracks, so that you could let your "master" branch (the one that's created automatically when you start a new repo) hold the stable version of your files, and you could create a "dev" branch (you can call it whatever you want, but "dev" [for "development"] is common) for stuff you're working on that isn't yet stable. Once you reach where you want to be, you can then "merge" the dev branch into the master one, effectively copying the changes from the experimental version into the main one. Becca Parker's tutorial about GitHub discusses this at http://newtfire.org/dh/explainGitShell.html#branch; it's written for the command-line GitHub client (not the graphical [GUI] client that you may be using), but it explains the principles in a way that applies across the board, and the GUI provides an easy way to create and manage branches.

In my own work, if I'm editing a document and don't plan to look at or return to the old version, I just edit it in the master branch (which is your only branch unless you've created others) and don't think about making explicit backups. If, though, I'm doing something substantially new and I'm not confident that it's where I want to go, I'll create a dev branch for it, so that I don't muck up the master one. One guideline is that the master branch should always be stable, so if you're making changes in a large project with multiple interacting pieces and changing one could break the whole business until you bring everything else into sync with it, you should do that in a dev branch, and merge the changes into the master branch only once you've regained stability.
